### ISSUE TITLE
C# Master leaks memory when calling Shutdown() and re-creating.

### ISSUE THREAD

**Issue by mprync** [dnp3/opendnp3#214]
This issue occurs when you create a Master, initialize it with points and connect. Then, close the connection, shut down the master by calling Shudown(); This will cause the Garbage Collector to not clean up the memory since it calls a finalizer on .NET objects.

The issue is due to the .NET Wrapper (DNP3ManagerAdapter) does not implement a finalizer.

Proposed fix:

DNP3ManagerAdapter.cpp
```javascript
DNP3ManagerAdapter::!DNP3ManagerAdapter()
{
   delete manager;
}

DNP3ManagerAdapter::~DNP3ManagerAdapter()
{
   this->!DNP3ManagerAdapter();
}
```

DNP3ManagerAdapter.h
```javascript
DNP3ManagerAdapter(System::Int32 concurrency, ILogHandler^ logHandler);
~DNP3ManagerAdapter();
!DNP3ManagerAdapter();

...
```

I haven't checked but if there needs to be cleanup done anywhere else in the wrapper, then this is the pattern that needs to be applied or the Garbage Collector tends to ignore it completely, causing a leak.

**Comment by jadamcrain** [comment]
Good catch @InFaMoUsZero.  We had a similar issue in one of the outstation adapter components recently. Want to submit a PR? If not, no biggie, I'll comb the issue list before the next RC.

**Comment by mprync** [comment]
No worries, I've added a pull request #215 with these changes.

**Comment by jadamcrain** [comment]
Resolved in f2eaaad736f8469b01916de01ee44fadf80fa1d5
